pkg/server/convert.go

```diff
@@ -19,28 +19,35 @@ TranslationFromData converts in specific schema serialized data to model.Transla
 TODO: Add support for converting non original, but already translated messages and mark them as TRANSLATED.
 */
 func TranslationFromData(params *uploadParams) (*model.Translation, error) {
-	var from func([]byte, bool) (model.Translation, error)
+	var (
+		err         error
+		original    bool
+		translation model.Translation
+	)
+
+	if params.original != nil {
+		original = *params.original
+	}
 
 	switch params.schema {
 	case translatev1.Schema_ARB:
-		from = convert.FromArb
+		translation, err = convert.FromArb(params.data, original)
```

Comment:
We can keep it as it was before, let `From<std>` determine the `original` from the file if it can.

And simply do once in this function body, right?

```go
  if params.original != nil {
    translation.Original = *params.original
  }
```

Reply:
I can see we passed the original to the `From<std>`. Do we have to? Does it impact the flow and processing of the file?

Reply:
Ignore my comment. I looked at he code and the `original` affects processing the translation file. 

Do https://github.com/translate-agent/translate/pull/204/files#r1400453299